### PR TITLE
[SD-5553] Modified saved search end-point.

### PR DESCRIPTION
**Pull request by nrvikas**

### CODE REVIEW

File: apps/saved_searches/saved_searches.py [hunk 1]
```diff
@@ -199,6 +201,19 @@ def on_fetched(self, docs):
...
+    def on_delete(self, doc):
+        self._validate_user(doc['user'])
+
+    def _validate_user(self, doc_user_id, create=False):
+        session_user = get_user(required=True)
+        if create and str(session_user['_id']) != doc_user_id:
+            raise SuperdeskApiError.forbiddenError(
+                'Unauthorized to create another user\'s search.')
+
+        if str(session_user['_id']) != doc_user_id and not current_user_has_privilege('global_saved_searches') and \
```
Comment (akintolga) [inline apps/saved_searches/saved_searches.py]: `current_user_has_privilege` function checks if the user is admin so we don't need to check it here again...

File: apps/saved_searches/saved_searches.py [hunk 2]
```diff
@@ -102,8 +102,9 @@ def on_fetched(self, docs):
 class SavedSearchesService(BaseService):
     def on_create(self, docs):
         for doc in docs:
-            if 'user' not in doc and request:
-                doc['user'] = request.view_args.get('user')
+            if 'user' not in doc:
```
Comment (akintolga) [inline apps/saved_searches/saved_searches.py]: Maybe 'user' could be removed from the resource as we can infer the user when creating a saved search for the request. Also when update and delete we wouldn't need the user. In this case we can save the user field before committing the changes to database.

File: features/saved_searches.feature [hunk 3]
```diff
@@ -232,29 +253,30 @@ Feature: Saved Searches
...
         {
         "name": "basket ball",
         "filter": {"query": {"q": "basket ball", "repo": "ingest"}},
-        "description": "abc"
+        "description": "abc",
+        "user": "#CONTEXT_USER_ID#"
         }
         """
-        When we patch "/users/#users._id#/saved_searches/#saved_searches._id#"
+        When we patch "/saved_searches/#saved_searches._id#"
```
Comment (akintolga) [inline features/saved_searches.feature]: Before this check the other user needs to be switched...

Comment (akintolga) [inline features/saved_searches.feature]: Also this test should pass if the updated search is global and fail if not.

Comment (nrvikas) [inline features/saved_searches.feature]: Updated. And also added other relevant test cases

File: apps/saved_searches/saved_searches.py [hunk 4]
```diff
@@ -102,8 +107,7 @@ def on_fetched(self, docs):
 class SavedSearchesService(BaseService):
     def on_create(self, docs):
         for doc in docs:
-            if 'user' not in doc and request:
-                doc['user'] = request.view_args.get('user')
+            doc['user'] = get_user(required=True)['_id']
```
Comment (petrjasek) [inline apps/saved_searches/saved_searches.py]: here you could use `get_user_id`

Comment (nrvikas) [inline apps/saved_searches/saved_searches.py]: Yup, I have modified the file. Thanks.

**Pull request merged**